tests/Microsoft.Identity.Test.Unit/MetricsTests.cs

```diff
@@ -159,6 +167,23 @@ private async Task TestAcquireTokenSilent_Async(PublicClientApplication pca, int
             Assert.AreEqual(expectedTokensFromCache, Metrics.TotalAccessTokensFromCache);
             Assert.AreEqual(expectedTokensFromBroker, Metrics.TotalAccessTokensFromBroker);
             Assert.IsTrue(Metrics.TotalDurationInMs > 0);
+
+            if (refreshIn)
+            {
+                //Force Refresh In
+                var at = pca.UserTokenCacheInternal.Accessor.GetAllAccessTokens().Single();
```

Comment:
Ok, so I had a look at the test and there seems to be a bug when we compute this field. Namely, if the token comes from the cache, the "reason" should be `None`, but instead it is currently `RefreshIn`. This is not a problem for telemetry, because we won't actually send it (no call to the /token endpoint).

Let's make sure (via tests) all 3 of our flows that hit the cache emit the correct flag.